cyclonedx-lib/sign_src/TemurinSignSBOM.java

```diff
@@ -0,0 +1,183 @@
+
+import org.webpki.json.JSONAsymKeySigner;
+import org.webpki.json.JSONObjectReader;
+import org.webpki.json.JSONSignatureDecoder;
+import org.webpki.json.JSONCryptoHelper;
+import org.webpki.json.JSONAsymKeyVerifier;
+import org.webpki.json.JSONObjectWriter;
+import org.webpki.json.JSONOutputFormats;
+import org.webpki.json.JSONParser;
+import org.webpki.util.PEMDecoder;
+
+import java.io.StringReader;
+import java.io.IOException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.GeneralSecurityException;
+import java.security.KeyPair;
+import java.security.PublicKey;
+import org.cyclonedx.exception.ParseException;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public final class TemurinSignSBOM {
+
+    private static boolean verbose = false;
+
+    // Create a logger for the class
+    private static final Logger LOGGER = Logger.getLogger(TemurinSignSBOM.class.getName());
+
+    private TemurinSignSBOM() {
+    }
+
+    /**
+     * Main entry.
+     * @param args Arguments for sbom operation.
+     */
+    public static void main(final String[] args) throws Exception {
+        String cmd = null;
+        String privateKeyFile = null;
+        String publicKeyFile = null;
+        String fileName = null;
+        boolean success = false; // add a new boolean success, default to false
+
+    for (int i = 0; i < args.length; i++) {
+        if (args[i].equals("--jsonFile")) {
+            fileName = args[++i];
+            success = true; // set success to true
+        } else if (args[i].equals("--privateKeyFile")) {
+            privateKeyFile = args[++i];
+            success = true; // set success to true
+        } else if (args[i].equals("--publicKeyFile")) {
+            publicKeyFile = args[++i];
+            success = true; // set success to true
+        } else if (args[i].equals("--signSBOM")) {
+            cmd = "signSBOM";
+            success = true; // set success to true
+        } else if (args[i].equals("--verifySignature")) {
+            cmd = "verifySignature";
+            success = true; // set success to true
+        } else if (args[i].equals("--verbose")) {
+            verbose = true;
+            success = true; // set success to true
+        }
+    }
+
+    if (!success) {
+        System.exit(1);
+    }
+
+    if (cmd.equals("signSBOM")) {
+        Bom bom = signSBOM(fileName, privateKeyFile);
+        if (bom != null) {
+            if (!writeJSONfile(bom, fileName)) {
+                System.exit(1);
+            }
+        } else {
+            System.exit(1);
+        }
+    } else if (cmd.equals("verifySignature")) {
+        success = verifySignature(fileName, publicKeyFile); // set success to the result of verifySignature
+        boolean isValid = success;
+        System.out.println("Signature verification result: " + (isValid ? "Valid" : "Invalid"));
+            if (!isValid) {
```

Comment:
remove this if, success already set above ^

Reply:
> remove this if, success already set above ^

@andrew-m-leonard then when should the system exit?